lib/core/modules/isar_module.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:io';
+
+import 'package:injectable/injectable.dart';
+import 'package:isar/isar.dart';
+import 'package:movna/data/models/activity_model.dart';
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+
+const baseIsar = Named('baseIsar');
+const baseIsarDirectory = Named('isarDirectory');
+
+@module
+abstract class IsarModule {
+  /// Factory method that creates an [Isar] instance from the [isarDirectory]
+  @baseIsar
+  Isar getBaseIsarInstance(@baseIsarDirectory Directory isarDirectory) {
+    final isar = Isar.getInstance(baseIsar.name!);
+    if (isar != null) {
+      return isar;
+    }
+
+    final newIsarInstance = Isar.openSync(
+      [ActivityModelSchema],
+      directory: isarDirectory.path,
+      name: baseIsar.name!,
+    );
+
+    return newIsarInstance;
+  }
+
+  /// Pre resolved factory that returns the directory for the [Isar] database.
+  @baseIsarDirectory
+  @preResolve
+  Future<Directory> get isarDirectory async {
+    final appDirectory = await getApplicationDocumentsDirectory();
+    return Directory(path.join(appDirectory.path, 'isar'));
+  }
```

Comment:
this will be initialized at app startup but I think if the app can't retrieve the documentsDirectory there no point to even try to display anything other than an error screen saying `Well something fucked up so restart the app please`.
So we could just `try-catch` the `configureDependencies` method in `main` and if it fails display said error message.
If it succeeds we'll launch the app normally